src/api/operator/utils.ts

```diff
@@ -49,3 +48,100 @@ export function getOrderStatus(order: RawOrder): OrderStatus {
     return 'open'
   }
 }
+
+/**
+ * Get order filled amount, both as raw amount (in atoms) and as percentage (from 0 to 1)
+ *
+ * @param order The order
+ */
+export function getOrderFilledAmount(order: RawOrder): { amount: BigNumber; percentage: BigNumber } {
+  let executedAmount, totalAmount
+
+  if (order.kind === 'buy') {
+    executedAmount = new BigNumber(order.executedBuyAmount)
+    totalAmount = new BigNumber(order.buyAmount)
+  } else {
+    executedAmount = new BigNumber(order.executedSellAmount).minus(order.executedFeeAmount)
+    totalAmount = new BigNumber(order.sellAmount)
+  }
+
+  const percentage = executedAmount.isZero() ? executedAmount : executedAmount.div(totalAmount)
```

Comment:
is it safe to divide from zero in `BigNumber`? then this could just be 
```ts
const percentage = executedAmount.div(totalAmount)
```
as it will still be zero

Reply:
It's safe, but this is to avoid returning NaN or Infinity
When we know it's 0, just return 0 and avoid issues later on when displaying it

Reply:
Well actually, I ran the tests with your suggestion and it still works.
Either I don't test the case I had in mind or it works as you describe :)

Will update